rllib/agents/trainer.py

```diff
@@ -927,16 +982,105 @@ def step(self) -> ResultDict:
                 # Allow logs messages to propagate.
                 time.sleep(0.5)
                 raise e
+
+            # Stopping criteria.
```

Comment:
why introduce stopping criteria with this PR for only _disable_distributed_execution_api case?

Reply:
Because the `execution_plan` handles this by itself via the `StandardMetricsReporting` op at the end. One of the reasons users didn't understand what the `min_iter_time_s` and `timesteps_per_iteration` config params meant and were doing under the hood.

Reply:
oh yes.
let's make this a util function in metric_ops.py, and see if we can pretty much reuse it in all the trainers?

Reply:
Totally! We should containerize these functionalities more.